pkg/api/types.go

```diff
@@ -822,6 +822,10 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// PullSecretNames is an optional list of secret names to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	PullSecretNames []string `json:"pullSecretNames,omitempty" description:"list of secret names available for pulling the container images"`
```

Comment:
@smarterclayton brought up the question of whether there was value in having this on a pod at all.  See https://github.com/deads2k/kubernetes/commit/e36818ac2ec1af0c5dd24925c5f4b0f90c778ef1 .  I don't feel very strongly.  I thought having it on a pod would be convenient (and it certainly was for testing), but I hadn't really considered whether it was essential.  We could force someone to simply use a service account.